docs/objective-c_abi.md

```diff
@@ -0,0 +1,15 @@
+# Objective-C ABI
+
+There are several Objective-C ABIs available:
+
+* Apple legacy ABI (also know as fragile ABI) - used on macOS 32bit and the old
```

Comment:
Just so I have the link correct in my head, this is also known as the Nextstep abi?

Reply:
Yes. I've moved the work on the ABI documentation to a separate branch until it's ready [1]. The description in the separate branch is a bit more clear and thorough.

[1] https://github.com/jacob-carlborg/dmd/blob/objc_abi_docs/docs/objective-c_abi.md

Reply:
OK.

I've just checked, and gcc objc seems to support all of these ABIs.  I should see if I can re-use the existing code sometime in gdc.

Reply:
Before Apple used Clang, they used GCC with this ABI that is implemented here. I don't think anything has changed since then. I'm not sure if that was only supported in their own version of GCC or in the standard version as well.

Reply:
BTW, some code review would be nice 😉.